### ISSUE TITLE
Add a "TPOT lite" configuration dictionary

### ISSUE THREAD

**Issue by rhiever** [rhiever/tpot#384]
By default, TPOT uses many computationally-expensive operators. We should add a "TPOT lite" version that uses simpler and faster operators for users that want to explore complex pipelines but not complex models.

cc @weixuanfu2016

**Comment by weixuanfu2016** [comment]
![classifier_used_time_on_pmlb](https://cloud.githubusercontent.com/assets/21084970/25011989/13614244-203d-11e7-9e6c-1ee30aade726.png)
Here is a test for getting a general feeling about run time of operators in `TPOTClassifer` using all benchmarks in [pmlb](https://github.com/EpistasisLab/penn-ml-benchmarks). The run time for all operators is based on `cross_val_score` with `cv=3`. Note for classifiers in `sklearn.ensemble`, I set `n_estimators=500` and keep other parameters as default. For other classifiers, default settings was applied. For preprocessors and selectors, a pipeline was build with `GaussianNB` as second step for `cross_val_score`. The boxplot is made by filtered out the run time > 30 seconds for clear comparison

**Comment by rhiever** [comment]
Thanks @weixuanfu2016 --- this is very useful. Perhaps as an initial attempt, TPOT lite should include all the operators starting from the bottom up to (and including) `SelectFromModel`, excluding `PolynomialFeatures` (as PF causes exponential feature growth)?

It's somewhat surprising that `SelectFromModel` is faster than `ExtraTreesClassifier`, as `SelectFromModel` uses the XTC right? Is it using fewer trees?

**Comment by weixuanfu2016** [comment]
Yep, `ExtraTreesClassifier` in `SelectFromModel` used default parameters (`n_estimators=10`), since we did not specify the `n_estimators` in [classifier dictionary in TPOT](https://github.com/rhiever/tpot/blob/master/tpot/config_classifier.py#L201-L204)

**Comment by rhiever** [comment]
I see. We should change the XTC to have at least 100 estimators in `SelectFromModel`, and in that case, it's probably too slow for TPOT lite.

**Comment by rhiever** [comment]
Oh, I see that we already made the change I mentioned above on the dev branch: https://github.com/rhiever/tpot/blob/development/tpot/config_classifier.py#L207

**Comment by weixuanfu2016** [comment]
Oh, did not see that. I will test it again with setting `n_estimators=100` instead of 500 in all `sklearn.ensemble` operators.

**Comment by weixuanfu2016** [comment]
![classifier_used_time_on_pmlb_n_esitmater_100](https://cloud.githubusercontent.com/assets/21084970/25044250/a2bec9dc-20f3-11e7-8ef3-a6089f8b2054.png)

`SelectFromModel` takes longer time with `n_estimators=100` in XTC, but it is still slightly faster than XTC only in this run. Maybe it is because the random seed is not fixed in this test.

**Comment by weixuanfu2016** [comment]
![regressor_used_time_on_pmlb_n_esitmater_100](https://cloud.githubusercontent.com/assets/21084970/25046465/9b734b24-20ff-11e7-93c9-151b44c07855.png)

Note: this figure above are generated using 89 benchmarks from openML. Just for a feeling about run time of these operators. More tests are running over thousands of benchmarks from openML, which should take a while to finish. I will update the figure after all tests are done.

For testing operators in TPOTRegressor, I used regression benchmarks from openML instead. Like what I did with operators in TPOTClassifier. The run time for all operators is based on cross_val_score with cv=3. Note for regressors in sklearn.ensemble, I set n_estimators=100 and keep other parameters as default. For other regressors, default settings was applied. For preprocessors and selectors, a pipeline was build with LinearRegression as second step for cross_val_score. The boxplot is made by filtered out the run time > 50 seconds for clear comparison.

**Comment by rhiever** [comment]
Great work. It looks like we should use all the operators up to (and including) PCA (but excluding PolynomialFeatures) for the TPOT lite classification dictionary. And up through (and including) Binarizer (but excluding PolynomialFeatures) for TPOT lite regression. @weixuanfu2016, can you please PR a demo of these configurations so we can test them out on the dev branch?

**Comment by weixuanfu2016** [comment]
OK, I will post a PR for the TPOT lite
